compiler/src/iree/compiler/Dialect/Stream/Transforms/EmplaceAllocations.cpp

```diff
@@ -54,18 +54,27 @@ static void replaceUsesAndTransfer(Value oldValue, Value newValue) {
   oldValue.replaceAllUsesWith(transferValue);
 }
 
+// TODO(#14566): multiple results with sparse ties don't work due to
+// implicit operand/result ordering on the dispatch ops. Flow and stream
+// dispatch ops and the executable entry points need to be reworked to
+// remove the implicit ordering. For now we only emplace results until the
+// first we can't then bail and leave them out-of-place.
 static bool tryEmplaceDispatchOp(IREE::Stream::AsyncDispatchOp dispatchOp) {
   bool didChange = false;
   for (auto [resultIndex, result] : llvm::enumerate(dispatchOp.getResults())) {
     // Ignore results with multiple users. We could potentially place these but
     // that makes tracking much more complicated.
-    if (!result.hasOneUse())
-      continue;
+    if (!result.hasOneUse()) {
+      // TODO(#14566): continue if sparse emplacement on multiple results.
```

Comment:
This is multi-use results which is a different case.

Reply:
the workaround here is to only process results until the first that can't be emplaced, so all conditions that may cause a result not to be emplaced need to bail